### PR TITLE
Add run parameters to huggingface callback

### PR DESCRIPTION
**Pull request by SSamDav**
This PR adds custom run parameters to the Huggingface Callback.
It tries to solve this [feature](https://github.com/aimhubio/aim/issues/1116) request.

### CODE REVIEW

**Review by gorarakelyan — APPROVED** [review]
Looks great! Congrats on the first contribution! 🚀 
Left one comment re key name of custom params. Thanks a lot for your time and effort.

File: aim/sdk/adapters/hugging_face.py [hunk 1]
```diff
@@ -44,6 +46,8 @@ def setup(self, args, state, model):
 
                 combined_dict = {**args.to_sanitized_dict()}
                 self._run['hparams'] = combined_dict
+                if self._run_params:
+                    self._run['params'] = self._run_params
```
Comment (gorarakelyan) [inline aim/sdk/adapters/hugging_face.py:50]: What about naming the params key as `run_params` as well?

**Comment by alberttorosyan** [comment]
@SSamDav thanks for contributing to Aim!
We just added a fix to our GH actions pipeline and checks are now triggered on fork PRs as well. May I ask you to pull latest `main` into you branch?
Thank you! 🙏

**Comment by SSamDav** [comment]
Ok, I made all the changes suggested! I just have one more question, I had one more feature request that I wanted to make, or implement in this case 😅 , but I don't know where to look in the code. Basically, I wanted to add more filter options like `run_param.name startswith`, `endswith`, `contains`. Could you help me identify where I could make this change?

**Comment by gorarakelyan** [comment]
@SSamDav I just tested it and magic! ✨ It is working out of the box thanks to executing queries in RestrictedPython context. Look at the example [here](http://play.aimstack.io:10001/runs?search=2E72oFqLmkAcicTCtQWYrtSkpEALU4dzDHWbia3qaKhD1YCVRxZwpMBkQC2TQBq4vrdUVvRQB4bVVomeBhuY6enarnTGEdZ&select=u7VyT9zYcD14L4M78TvQxQMPCnZQjy3QmRkjdjqjNiYPDQAVixc4BmGor4EogEMMp2QBuonW1vZ1RJHAgXp8kR1i5JgKTb5o8Sn8H18NtKjPUCeNGsz8s99vqKeNDgsnq1r2FHduUghcWNeWa1tw4mYQdbV62eKbEoDajyCiwJZPQJWZT5pr3nHETZE8n6td5jL9gtomEFbusbBAiMaR7p1ypE3oamMSfT688pSXhrFYNsmee7Y2XJzcTpHpq5eNNWugYgjqXJvRq6e1tRn3tGU8do3gtavG).
What other python builtin functions would you like to use?

**Comment by gorarakelyan** [comment]
@alberttorosyan could you please help us here? I see some tests continue to fail.

**Comment by SSamDav** [comment]
Ohh thanks I didn't find anything in documentation so I thought It was not implemented 😅  

Hmmm, the only thing that comes to mind is filtering using dates, but probably you also have that implemented!

**Comment by gorarakelyan** [comment]
@SSamDav hm that's weird, probably some issues with the variable representation. I guess converting to string before applying `startswith` should work, like `str(run.hparams.align).startswith('XL')` 😊 

@mihran113 could you please check why `run.hparams.align.startswith('XL')` doesn't work, but `str(run.hparams.align).startswith('XL')` works? 🤔

**Comment by gorarakelyan** [comment]
@SSamDav re filtering using dates: you've guessed it right! 
Filter runs by creation_time e.g. `run.creation_time >= <unix_timestamp>`. But I believe there could be a better interface for filtering with dates. Maybe you could share any ideas on this? :)

**Comment by SSamDav** [comment]
`str(run.hparams.align).startswith('XL') works?` yes it works thks!!!

**Comment by SSamDav** [comment]
>  Maybe you could share any ideas on this? 
Regarding this, after knowing how it's supposed to do, it seems pretty standard. I think it is the same way you do in pandas.

**Comment by alberttorosyan** [comment]
@SSamDav @gorarakelyan just pushed a fix for running actions on forked repository PRs.
@SSamDav, may I ask you to sync with `main` branch once again? Thank you! 🙏

**Comment by alberttorosyan** [comment]
regarding this:

> run.hparams.align.startswith('XL') doesn't work, but str(run.hparams.align).startswith('XL') works

This is a known limitation of Aim queries. It is a result of how `.` (`__getattr__`) is treated in the context of query execution. 
@SSamDav if you find this limitation a major US issue, could you please open a Github issue with enhancement proposal?

**Comment by gorarakelyan** [comment]
Merged the PR to avoid conflicts. @SSamDav thanks for the first contribution!

**Comment by SGevorg** [comment]
@SSamDav thanks for the contribution. Awesome stuff !! 🙌 😊

**Pull request merged**